src/lib/services/project-service.ts

```diff
@@ -778,12 +784,32 @@ export default class ProjectService {
             },
         ]);
 
-        const currentWindowTimeToProdReadModel = new TimeToProduction(
+        // todo: remove after release of the improved query
+        const timeToProduction = new TimeToProduction(
             allFeatures,
             productionEnvironments,
             eventsData,
         );
 
+        const avgTimeToProdCurrentWindowFast =
+            TimeToProduction.calculateAverageTimeToProd(
+                await this.projectStatsStore.getTimeToProdDates(projectId),
+            );
+        const avgTimeToProdCurrentWindowSlow =
+            timeToProduction.calculateAverageTimeToProd();
+
+        const avgTimeToProdCurrentWindow = this.flagResolver.isEnabled(
+            'projectStatusApiImprovements',
```

Comment:
I see you're not using variants, OK, new flag is easier to understand

Reply:
I found that flagResolver interface doesn't expose variants